app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanViewModelTest.kt

```diff
@@ -0,0 +1,390 @@
+        val fakeSessionFormat = mock<SimpleSessionFormat> {
+            on { format(any<Session>(), anyOrNull()) } doReturn "session-61"
+        }
+        val viewModel = createViewModel(repository, simpleSessionFormat = fakeSessionFormat)
+        viewModel.share(Session("61"))
+        assertLiveData(viewModel.shareSimple).isEqualTo("session-61")
+        verifyInvokedOnce(repository).readMeta()
+    }
+
+    @Test
+    fun `shareToChaosflix posts to shareJson property`() {
+        val repository = createRepository()
+        val fakeSessionFormat = mock<JsonSessionFormat> {
+            on { format(any<Session>()) } doReturn "session-62"
+        }
+        val viewModel = createViewModel(repository, jsonSessionFormat = fakeSessionFormat)
+        viewModel.shareToChaosflix(Session("62"))
+        assertLiveData(viewModel.shareJson).isEqualTo("session-62")
+    }
+
+    @Test
+    fun `selectDay updates preserveVerticalScrollPosition property when displayDayIndex changes`() {
+        val repository = createRepository(displayDayIndex = 1)
+        val viewModel = createViewModel(repository)
+        viewModel.preserveVerticalScrollPosition = true
+        viewModel.selectDay(dayItemPosition = 3)
+        assertThat(viewModel.preserveVerticalScrollPosition).isFalse()
+        verifyInvokedOnce(repository).readDisplayDayIndex()
+        verifyInvokedOnce(repository).updateDisplayDayIndex(displayDayIndex = 4)
+    }
+
+    @Test
+    fun `selectDay never updates preserveVerticalScrollPosition property when displayDayIndex is the same`() {
+        val repository = createRepository(displayDayIndex = 4)
+        val viewModel = createViewModel(repository)
+        viewModel.preserveVerticalScrollPosition = true
+        viewModel.selectDay(dayItemPosition = 3)
+        assertThat(viewModel.preserveVerticalScrollPosition).isTrue()
+        verifyInvokedOnce(repository).readDisplayDayIndex()
+        verifyInvokedNever(repository).updateDisplayDayIndex(displayDayIndex = any())
+    }
+
+    @Test
+    fun `saveSelectedDayIndex invokes repository function`() {
+        val repository = createRepository()
+        val viewModel = createViewModel(repository)
+        viewModel.saveSelectedDayIndex(dayIndex = 2)
+        verifyInvokedOnce(repository).updateDisplayDayIndex(displayDayIndex = 2)
+    }
+
+    @Test
+    fun `scrollToCurrent never posts to scrollToCurrentParameter property when sessions is empty`() {
+        val scheduleData = ScheduleData(0, emptyList())
+        val repository = createRepository(uncanceledSessionsForDayIndexFlow = flowOf(scheduleData))
+        val viewModel = createViewModel(repository)
+        viewModel.scrollToCurrent()
+        verifyInvokedNever(repository).readDisplayDayIndex()
+        assertLiveData(viewModel.scrollToCurrentParameter).isNull()
+    }
+
+    @Test
+    fun `scrollToCurrent posts to scrollToCurrentParameter property when session is present and day indices match`() {
```

Comment:
This fails now locally - can you confirm?
nerd.tuxmobil.fahrplan.congress.schedule.FahrplanViewModelTest > scrollToCurrent posts to scrollToCurrentParameter property when session is present and day indices match FAILED
    expected: ScrollToCurrentParameter(scheduleData=ScheduleData(dayIndex=3, roomDataList=[RoomData(roomName=, sessions=[nerd.tuxmobil.fahrplan.congress.models.Session@8404c68a])]), dateInfos=[DateInfo(dayIndex=3, date=2021-12-17T00:00:00Z)])
    but was : null
        at app//nerd.tuxmobil.fahrplan.congress.schedule.FahrplanViewModelTest.scrollToCurrent posts to scrollToCurrentParameter property when session is present and day indices match(FahrplanViewModelTest.kt:299)

Reply:
:thinking: I have run all tests in this class and this particular test multiple times. All tests pass. I am using Android Studio Bumblebee 2021.1.1 Beta 5. - Can you please try again?

Reply:
It looks like Gradle was the producer of this issue - after clean out all the caches...it runs green...interesting though, since BuildCaches should'nt be active.

Reply:
@bitPogo Just for you to know. The same error appeared at Travis CI now once. The test seems to be flaky. It succeeded when I re-run the test on CI. If someone of us has time we can have a look at it.